spec/unit/filecache_spec.rb

```diff
@@ -84,6 +84,36 @@ def write_file(file, contents)
         end
       end
 
+      it "sets the encoding to UTF-8 when reading a file" do
+        begin
+          original_encoding = Encoding.default_external
+          Encoding.default_external = Encoding::ISO_8859_1
+
+          Dir.mktmpdir do |dir|
+            file = File.join(dir, "testing")
+            write_file(file, "my data")
+            expect(@cache.read_file(file).encoding).to eq(Encoding::UTF_8)
+          end
+        ensure
+          Encoding.default_external = original_encoding
+        end
+      end
+
+      it "reads a file with unicode characters" do
+        begin
+          original_encoding = Encoding.default_external
+          Encoding.default_external = Encoding::ISO_8859_1
+
+          Dir.mktmpdir do |dir|
+            file = File.join(dir, "testing")
+            write_file(file, "\u2603")
+            expect(@cache.read_file(file)).to eq("\u2603")
+          end
```

Comment:
As a point of note, in the existing code ... the file bytes are actually read correctly, but they are not considered the appropriate encoding:


```
[1] pry(#<RSpec::ExampleGroups::HieraFilecache::ReadFile>)> foo = @cache.read_file(file)
=> "â\x98\x83"
[2] pry(#<RSpec::ExampleGroups::HieraFilecache::ReadFile>)> foo.bytes
=> [226, 152, 131]
[3] pry(#<RSpec::ExampleGroups::HieraFilecache::ReadFile>)> bar = foo.dup.force_encoding(Encoding::UTF_8)
=> "\u2603"
```